README.md

```diff
@@ -10,7 +10,118 @@ both in CI environments and in command-line to produce a tile.
+groups.
+
+Kiln verifies that the checksum (SHA1) of the downloaded release matches
+checksum specified for the release in the assets lock file. If the checksums do
+not match, then the releases that don't match will be deleted from disk. *Since
+BOSH releases from different directors with the same packages result in complied
+releases with different hashes this may result in some problems for local
+development.*
+
+Kiln will not download releases if an existing release exists with the correct
+release version and checksum.
+
+#### Assets File
+The assets file must also have information about how to access the S3 Bucket.
+The following are **required** in the `compiled_releases` hash in an assets
+file.
+
+- `type`: must be "s3"
+- `bucket`: must be the name of the s3 bucket
+- `region`: must be the region of the bucket
+- `access_key_id`: must be an IAM access key id that has read permission for the
+  specified bucket
+- `secret_access_key`: must be the secret for the specified `access_key_id`
+- `regex:`: is used to specify a capture groups to map release and stemcell
+  data to an object in the S3 bucket. The following capture groups must be in
+  the regex. The regex capture group maps to fields in the assets lock file.
+  **All of the following capture groups must be in the RegEx specified in the
+  assets file**
+  - `release_name` must map to the assets.lock file under `releases[].name`
+  - `release_version` must map to the assets.lock file under
+    `releases[].version`
+  - `stemcell_os` must map to the assets.lock file under `stemcell_criteria.os`
+  - `stemcell_version` must map to the assets.lock file under
+    `stemcell_criteria.version`
+
+### Assets Lock
+
+Currently the releases for assets lock file is can not be generated by kiln.
+The update command is in development and only (loosely) supports updating the
+stemcell based on stemcells on https://network.pivotal.io. On PAS Release
+Engineering we use a task in our CI to generate this file.
+
+The file has two top level members `releases` and `stemcell_criteria`.
+
+The `releases` member is an array of members with each element having the
+having the following members.
+- `name`: bosh release name
+- `sha1`: checksum of the tarball
+- `version`: semantic version of the release
+
+The `stemcell_criteria ` member is an array of members with each element having the
+having the following members.
+- `name`: bosh release name
+- `sha1`: checksum of the tarball
+- `version`: semantic version of the release
+
+### Example Variable Interpolation
+
+```
+$ cat assets.yml
+compiled_releases:
+ type: s3
+ bucket: compiled-releases
+ region: us-west-1
+ access_key_id: $(variable "aws_access_key_id")
+ secret_access_key: $(variable "aws_secret_access_key")
+ regex: ^2.6/.+/(?P<release_name>[a-z-_0-9]+)-(?P<release_version>v?[0-9\.]+)-(?P<stemcell_os>[a-z-_]+)-(?P<stemcell_version>\d+\.\d+)(\.0)?\.tgz$
+```
+
+*Storing credentials like S3 keys in a repo is probably not the best idea, so
```

Comment:
I think we can use even stronger language: "Credentials like S3 keys are not stored in git repos.  To support separating that information from non-sensitive configuration, you can reference variables like you do in tile config."